packages/nimble-components/src/table-column/text-base/index.ts

```diff
@@ -18,3 +18,12 @@ export abstract class TableColumnTextBase extends mixinGroupableColumnAPI(
         this.columnInternals.operandDataRecordFieldName = this.fieldName;
     }
 }
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinTextBase<TColumnConfig>() {
+    return mixinGroupableColumnAPI(
+        mixinFractionalWidthColumnAPI(
+            mixinColumnWithPlaceholderAPI(TableColumnTextBase<TColumnConfig>)
```

Comment:
Is it possible to take the class to mixin on as a parameter? i.e. something with type like `TBase extends TableColumnTextBase<TColumnConfig>`

My reasons are this method isn't really a "mixin" since it's not mixing in behavior on a provided class. It's more creating a TextBase class vs a mixin of behavior onto an exisiting class.

Other reason is that we don't need it now but if we wanted to have additional class layers this would be restrictive, i.e.

```
DateText -> TimeTextBase -> TextBase -> TableColumn
DurationText -> TimeTextBase -> TextBase -> TableColumn
NumberText -> TextBase -> TableColumn
```

We'd want to be able to do something like
```
DateText extends mixinTextBase(TimeTextBase) {}
DurationText extends mixinOther(mixinTextBase(TimeTextBase)) {}
NumberText extends mixinTextBase(TextBase) {}
```

Reply:
I figured out how to get the typing to work correctly with the approach you suggested. I've also tested that `mixinTextBase` works correctly with a type that inherits from `TextBase`, such as `TimeTextBase` in your example.